src/utils/poolData/balancerV1.ts

```diff
@@ -48,3 +61,33 @@ export async function getBalancerV1Liquidity(
 
   return response
 }
+
+async function getBalancerV2(tokenAddress: string, chainId: ChainId) {
+  let response: BalBalances = {
+    tokenBalance: BigNumber.from(0),
+    wethBalance: BigNumber.from(0),
+  }
+
+  // TODO: add balancer v2 factory abi
+
+  // TODO: fetch pool id for pair (via subgraph)
+  // https://dev.balancer.fi/resources/pool-interfacing#poolids
+  // https://thegraph.com/hosted-service/subgraph/balancer-labs/balancer-v2
+
+  // TODO: use vault.getPoolTokens() to obtain balances
+  // https://dev.balancer.fi/resources/pool-interfacing#pool-balances
+
+  return response
```

Comment:
Balancer on Polygon is a whole new version of the ABI and lots of things have changed compared to V1. As far as I got it from the docs this could be a way to get the necessary data. Does it seem like a good way?

(see links for docs)

Reply:
Wanted to clarify before wasting time on implementing. In case someone knows a better approach.

Reply:
Now that I think about it, I could probably just fetch all data from the subgraph, right? (while using it already)